server/src/main/java/org/elasticsearch/action/admin/indices/close/TransportVerifyShardBeforeCloseAction.java

```diff
@@ -109,9 +110,13 @@ private void executeShardOperation(final ShardRequest request, final IndexShard
         if (clusterBlocks.hasIndexBlock(shardId.getIndexName(), request.clusterBlock()) == false) {
             throw new IllegalStateException("Index shard " + shardId + " must be blocked by " + request.clusterBlock() + " before closing");
         }
-        indexShard.verifyShardBeforeIndexClosing();
-        indexShard.flush(new FlushRequest().force(true).waitIfOngoing(true));
-        logger.trace("{} shard is ready for closing", shardId);
+        if (request.isPhase1()) {
+            indexShard.sync();
```

Comment:
BWC can be tricky here. If the data nodes are on the latest version but the master node is still on an old version, we won't have the pre-flight request. Then the global checkpoint might not be up to date when we execute the sanity check (i.e., global checkpoint == max_seqno).

Reply:
I think that the worst that will happen is that the close index call fails. Note that this is only an issue with async translog syncing. In that case, the way to fix it is to issue a flush, which will bring the global checkpoint on the primary up to the max sequence number. You can then just run the close index call again. Also note that this could go into 7.2.1, so I'm not too concerned about this.